astropy/table/table.py

```diff
@@ -16,6 +16,7 @@
 
 from .. import log
 from ..io import registry as io_registry
+from ..io.fits.util import  _is_int
```

Comment:
There is no need for the `isint` function at this point with the `np.asarray` solution you have implemented.